### ISSUE TITLE
Model does not checkin

### ISSUE THREAD

**Issue by ejmulman** [opensourceBIM/BIMserver#958]
I am trying to checkin a model what does not have problems with tekla bimsight or solibri modelviewer.

Checkin error code;
[32_railing_hekwerken.ifc] Error on line 84369: Enum type IfcRampTypeEnum has no literal value 'ELEMENT'

Checkin with ifc2x3tc1 streaming (tried other serializers, also did not work)

It's an 30mb IFC file created form an solid works aplication and other viewers can read it.

Here the 84360 - 84375 lines from the IFC file;
#84360=IFCMEMBER('1mWRImLF549ROD1ZOYG$8q',#8,'2106978_151021 VOLLEDIG',$,'2106978_151021 VOLLEDIG',#410385,#400393,'B7B93CF1-1F60-404C-84-39-906D4A8464FC');
#84361=IFCMEMBER('0tVFAfj3XCzxTHNEVj4YwH',#8,'2107020_151021 VOLLEDIG',$,'2107020_151021 VOLLEDIG',#410482,#400394,'03130665-F96D-4C71-BB-05-FC7721883574');
#84362=IFCMEMBER('0XVA3VXoj4jR_hA8K1Riwe',#8,'2104125_148420 VOLLEDIG',$,'2104125_148420 VOLLEDIG',#410532,#400395,'0F4D1A78-0988-4803-96-23-AA394A361D57');
#84363=IFCRAMP('1p9mogdg59U9e7SGWic_Md',#4,'m16x190 draadeind a2_98863a420',$,'m16x190 draadeind a2_98863a420',#403128,#400396,'CD57CCA7-5DAB-41BC-B3-1D-BE1C1D69870B',.ELEMENT.);
#84364=IFCRAMP('0b$3tKvD5ExgJvz1_czGJU',#4,'2104152',$,'2104152',#403193,#400397,'8A085E98-0601-403F-90-6F-73C76349D188',.ELEMENT.);
#84365=IFCRAMP('09IeVvwTT7SetkVduXkOED',#4,'2108419',$,'2108419',#403296,#400398,'7B61E40E-65A7-4F46-85-48-0ABC07A8D1BD',.ELEMENT.);
#84366=IFCRAMP('065fLjfZz6_fVUqARNb1Wc',#4,'2101674',$,'2101674',#403473,#400399,'B1A393E9-DFD4-48FF-B1-52-E376C57240CC',.ELEMENT.);
#84367=IFCRAMP('3ATD2Lfn9Dmhvu245zM29R',#7,'m12 zesk.moer 0.8d a4 coat',$,'m12 zesk.moer 0.8d a4 coat',#404241,#400400,'C490FA7B-D5B5-446E-A8-67-4A2DD49B6DBE',.ELEMENT.);
#84368=IFCRAMP('3unGtkBRT3v8USInO7Nvqi',#7,'m8x18 tapb alu p60',$,'m8x18 tapb alu p60',#404712,#400401,'747E66BC-B55A-426C-AD-5A-85C7A4C30215',.ELEMENT.);
#84369=IFCRAMP('3z9cd_5DDFLOwhSr$94w8l',#7,'2102026',$,'2102026',#404716,#400402,'F3C939B7-A215-4EC0-86-D4-11D8D3CD24D8',.ELEMENT.);
#84370=IFCRAMP('2621nqup543gcHCQ2_hStN',#7,'2107006',$,'2107006',#404748,#400403,'69013647-FDEB-4CBA-80-A0-B6F33FE840A0',.ELEMENT.);
#84371=IFCRAMP('0JvQA6zRP00Q1c04kGbIO3',#6,'2107020_151021 VOLLEDIG',$,'2107020_151021 VOLLEDIG',#406187,#400404,'196F5C67-AE9B-48C4-B5-E3-D016CF55A94D',.ELEMENT.);
#84372=IFCRAMP('3kMcOuSG98xgnY8M2x0DI8',#6,'2108429_151021 VOLLEDIG',$,'2108429_151021 VOLLEDIG',#406229,#400405,'E2E55E1B-FAE4-4DD3-BD-13-41B393CE4E8C',.ELEMENT.);
#84373=IFCRAMP('0q4xCfudf75eSI0OWrhHoR',#5,'2107376_151022 VOLLEDIG',$,'2107376_151022 VOLLEDIG',#407796,#400406,'E14E4735-0EF2-4307-B1-E3-0AAD26614BF9',.ELEMENT.);
#84374=IFCRAMP('0t1UtKc351zeWINirrmETX',#9,'2106983',$,'2106983',#409131,#400407,'C28F7F02-7FF5-452E-B2-80-91D54DEDC1FF',.ELEMENT.);
#84375=IFCRAMP('0ljyAHHXP8GhEqCU0IDg$b',#9,'2107005_151021 VOLLEDIG',$,'2107005_151021 VOLLEDIG',#409153,#400408,'3CD2173E-37C7-45B0-85-FF-3F6F7D737DE7',.ELEMENT.);

**Comment by rubendel** [comment]
How much clearer do you want the error message to be? Have a look at the spec please. I am assuming your file is Ifc2x3tc1 since you didn't mention it.

http://www.buildingsmart-tech.org/ifc/IFC2x3/TC1/html/ifcsharedbldgelements/lexical/ifcramptypeenum.htm

http://www.buildingsmart-tech.org/ifc/IFC2x3/TC1/html/ifcsharedbldgelements/lexical/ifcramp.htm.

Please contact your authoring tool provider and have them fix the export.

**Comment by ejmulman** [comment]
I am sorry for my obvious mistake (I now understand), but I did not fully understand the error message. You pointed me in the right direction, thank you for your response. 

I recieved this file and could not process it. After an notepad edit change 15 lines IfcRamp changed ".ELEMENT." to ".NOTDEFINED." at least I could process and work it. 

So another thanks and I appreciate the work done for bimserver.
